src/configs/FirebaseConfig.js

```diff
@@ -59,6 +60,23 @@ const FirebaseConfig = (props) => {
     }
   };
 
+  const getUserProfileV2 = async (url) => {
+    if (url && typeof url === 'string') {
+      const userId = await getUserId();
+      const parts = url.split('/profile/');
+      const urlSplit = parts?.[1];
+      const paramSplit = urlSplit?.split('?userId=');
+      const username = paramSplit?.[0];
+      const targetUserId = paramSplit?.[1];
+      const data = {
+        other_id: targetUserId,
+        username,
+        user_id: userId
+      };
+      handleMovePage(USER, data);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getUserProfileV2` function has been introduced. It appears to be extracting user information from a URL string. While the logic seems correct, there are potential issues with error handling and data validation. 

1. If the URL does not contain '/profile/', `parts[1]` will be undefined, causing an error when trying to split it on line 68.
2. If the URL does not contain '?userId=', `paramSplit[1]` will be undefined, which might cause issues when it's assigned to `targetUserId` on line 70.
3. There's no check to ensure that `userId`, `username`, and `targetUserId` are valid before they're used to create the `data` object.

Consider adding checks for these conditions to prevent potential errors.

```diff
  const getUserProfileV2 = async (url) => {
    if (url && typeof url === 'string') {
      const userId = await getUserId();
      const parts = url.split('/profile/');
+       if (!parts[1]) return; // Add this line
      const urlSplit = parts?.[1];
      const paramSplit = urlSplit?.split('?userId=');
+       if (!paramSplit[1]) return; // Add this line
      const username = paramSplit?.[0];
      const targetUserId = paramSplit?.[1];
+       if (!userId || !username || !targetUserId) return; // Add this line
      const data = {
        other_id: targetUserId,
        username,
        user_id: userId
      };
      handleMovePage(USER, data);
    }
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->